sdk/python/kfp/dsl/_volume_op.py

```diff
@@ -68,6 +68,37 @@ def __init__(self,
                         if data_source is not one of (str, PipelineParam,
                             V1TypedLocalObjectReference)
         """
+        # condition when action == "delete"
+        if "action" in kwargs and kwargs["action"] == "delete":
+            if "k8s_resource" in kwargs:
+                if resource_name or size or storage_class or modes or annotations:
+                    raise ValueError("You cannot provide k8s_resource along with "
+                                     "other arguments.")
+                if not isinstance(kwargs["k8s_resource"], V1PersistentVolumeClaim):
+                    raise ValueError("k8s_resource in VolumeOp must be an instance"
+                                     " of V1PersistentVolumeClaim")
+                super().__init__(**kwargs)
+                return
+
+            # Set the k8s_resource
+            if not match_serialized_pipelineparam(str(resource_name)):
+                resource_name = sanitize_k8s_name(resource_name)
+
+            pvc_metadata = V1ObjectMeta(
+                name="{{workflow.name}}-%s" % resource_name,
+            )
+
+            k8s_resource = V1PersistentVolumeClaim(
+                api_version="v1",
+                kind="PersistentVolumeClaim",
+                metadata=pvc_metadata,
+            )
+            super().__init__(
+                k8s_resource=k8s_resource,
+                **kwargs,
+            )
+            return
```

Comment:
This part is duplicate with existing code base. Is it possible to reuse them? I am thinking it should be fine to pass `delete` along to `ResourceOp` and let `ResourceOp` handle the real logic. We don't necessarily to have a branch here to handle this case.

Reply:
For this part, I'm thinking about using an elegant way to avoid duplication and increase reusability. Will propose a new revision to see how it goes.